lib/ecto/repo/supervisor.ex

```diff
@@ -110,11 +110,33 @@ defmodule Ecto.Repo.Supervisor do
             password: password,
             database: database,
             hostname: info.host,
-            port:     info.port]
+            port:     info.port] ++ parse_uri_query(info)
```

Comment:
Optionally, we can get rid of `++` by passing opts as accumulator to the `parse_uri_query/1`.